File: twosum/files/TwoSuM_bruteforce.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for (int i = 0; i < nums.size(); i++) {
            int value = target - nums[i];
            for (int j = i + 1; j < nums.size(); j++) {
                if (nums[j] == value) {
                    return {i, j};
                }
            }
        }

        return {};
    }
};

void printVector(const vector<int>& values) {
    cout << "[";
    for (int i = 0; i < values.size(); i++) {
        cout << values[i];
        if (i != values.size() - 1) {
            cout << ", ";
        }
    }
    cout << "]\n";
}

int main() {
    Solution sol;

    vector<int> nums1 = {2, 7, 11, 15};
    vector<int> nums2 = {3, 3};
    vector<int> nums3 = {3, 2, 4};

    printVector(sol.twoSum(nums1, 9));
    printVector(sol.twoSum(nums2, 6));
    printVector(sol.twoSum(nums3, 6));

    return 0;
}

File: TwoSuM_bruteforce.py
from typing import List 
class Solution:
    def two_sum(self,nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            value = target - nums[i]
            for j in range(i+1,len(nums)):
                if nums[j] == value:
                    return [i,j]

sol = Solution()
print(sol.two_sum([2,7,11,15],9))
print(sol.two_sum([3,3],6))
print(sol.two_sum([3,2,4],6))